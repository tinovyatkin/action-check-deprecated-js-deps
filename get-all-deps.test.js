import getAllDeps from "./get-all-deps";

describe("getAllDeps", () => {
  it("returns map of all deps", () => {
    const res = getAllDeps();
    expect(res).toBeInstanceOf(Map);
    expect(res.size).toBe(19);
  });

  it("works with monorepo", () => {
    process.env.INPUT_ROOT = "__tests__/monorepo-fixture";
    const res = getAllDeps();
    expect(res).toMatchSnapshot();
  });
});
