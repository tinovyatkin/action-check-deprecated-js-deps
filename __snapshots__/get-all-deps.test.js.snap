// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getAllDeps works with monorepo 1`] = `
Map {
  "test-dev-dep" => [
    {
      "file": "packages/a",
      "type": "devDependencies",
      "version": "3.4.5",
    },
    {
      "file": "packages/b",
      "type": "devDependencies",
      "version": "3.4.5",
    },
  ],
  "test-dep" => [
    {
      "file": "packages/b",
      "type": "dependencies",
      "version": "latest",
    },
    {
      "file": "package.json",
      "type": "dependencies",
      "version": "1.2.3",
    },
  ],
  "test-resolution" => [
    {
      "file": "packages/b",
      "type": "resolutions",
      "version": "1.1.1",
    },
  ],
}
`;
